test/gen-template/matter/external-addon-helper.js

```diff
@@ -0,0 +1,31 @@
+/**
+ *
+ *    Copyright (c) 2021 Silicon Labs
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+// This is an example of an external addon helper for templates.
+
+async function test_external_addon_helper(api, context) {
+  //add queries here
+  return 'This is example of test external addon helper.'
+}
+function initialize_helpers(api, context) {
+  api.registerHelpers(
+    'test_external_addon_helper',
+    test_external_addon_helper.bind(null, api, context),
```

Comment:
I don't understand this. You are binding the function with `null` as this, and passing api and context. Why are you creating a bound function here? Why not just use the actual function? I assumed we will have a wrapper function here, not a bound instance of the same function...

Reply:
Ok I'll look into it, thanks